File: 2026/04-April/10-Friday/solution.c
#include <stdio.h>
#include <stdlib.h>
 
#define MAXN 101

int minimumDistance(int* nums, int numsSize) {
    int* pos[MAXN];
    int count[MAXN] = {0};

    // allocate memory
    for (int i = 0; i < MAXN; i++) {
        pos[i] = (int*)malloc(sizeof(int) * numsSize);
    }

    // store indices
    for (int i = 0; i < numsSize; i++) {
        int val = nums[i];
        pos[val][count[val]++] = i;
    }

    int ans = 1e9;

    for (int v = 0; v < MAXN; v++) {
        if (count[v] >= 3) {
            for (int i = 0; i + 2 < count[v]; i++) {
                int dist = 2 * (pos[v][i+2] - pos[v][i]);
                if (dist < ans) ans = dist;
            }
        }
    }

    // free memory
    for (int i = 0; i < MAXN; i++) {
        free(pos[i]);
    }

    return (ans == 1e9) ? -1 : ans;
}

File: 2026/04-April/10-Friday/solution.kt
class Solution {
    fun minimumDistance(nums: IntArray): Int {
        val map = mutableMapOf<Int, MutableList<Int>>()

        for (i in nums.indices) {
            map.getOrPut(nums[i]) { mutableListOf() }.add(i)
        }

        var ans = Int.MAX_VALUE

        for (list in map.values) { 
            for (i in 0 until list.size - 2) {
                ans = minOf(ans, 2 * (list[i + 2] - list[i]))
            }
        }

        return if (ans == Int.MAX_VALUE) -1 else ans
    }
}
